sim/core/apl.go

```diff
@@ -181,6 +181,12 @@ func (apl *APLRotation) DoNextAction(sim *Simulation) {
 }
 
 func (apl *APLRotation) getNextAction(sim *Simulation) *APLAction {
+	for _, unit := range sim.AllUnits {
```

Comment:
Looping through all units on the hot path will be a noticeable performance loss. Seems like we can just check `apl.unit` here anyway right?

Reply:
yep, thanks!

Reply:
I still see a loop over all units here, is there a missing commit?